driver-core/src/main/com/mongodb/client/model/IndexOptionDefaults.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model;
+
+import org.bson.BsonDocument;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
+
+/**
+ * The default options for a collection to apply on the creation of indexes.
+ *
+ * @since 3.2
+ * @mongodb.driver.manual reference/command/createIndexes Index options
+ * @mongodb.server.release 3.2
+ */
+public class IndexOptionDefaults implements Bson {
```

Comment:
I'm fine with this class, but it should not not implement Bson, as the need for it is an implementation detail of the driver and not something users of the class need be aware of.  And it doesn't help them since CreateCollectionOptions required IndexOptionDefaults anyway, and will not accept any Bson instance.

We can always retrofit it later to implement Bson if need be.